### DISCUSSION TITLE
I deleted all the files on my computer

### DISCUSSION THREAD

**Discussion by ghost** [community/community#51702] [New to GitHub]
I was on VS code and there was 10000+ commits waiting to be made I went to delete them. And when it said are you sure I said yes. Now every file on my computer is gone is there any way I can get them back?

**Comment by ghost** [comment]
Well,

If you do not have them in your recycle bin folder, then no- However, you technically can restore some files using a 3rd party software.

**Reply by ghost** [reply]
What kind of software?

**Reply by ghost** [reply]
For you to figure out.

Most of them are paid, some only give you X MB that you can restore for free.

**Comment by neutrinomuon** [comment]
Hello, I suppose you're using windows... There are several recovery software (but most of them paid):

- Recuva
- EaseUS Data Recovery Wizard
- Disk Drill
- Stellar Data Recovery

These are just a few examples, but there are many other data recovery software for Windows. It's important to note that data recovery is not always possible, especially on Windows :(! If you have access to Linux, you can use undelete (it works amazingly well).

One question: Have you tried `git fsck --lost-found` command? If not, it might work, however, I don't know a way to recover all of them automatically, you'll need to go one by one, and maybe automate a list:

1. Open the Git Bash terminal and go to the repository where the commits were deleted;
2. Type the following command to list all the deleted commits:
<pre>
git fsck --lost-found
</pre>
4. Look for the commit IDs in the output that you want to recover. I think they are listed as dangling commits;
5. Type the following command to recover a commit:
<pre>
git cherry-pick commit-id
</pre>
6. Replace 'commit-id' with the commit ID you want to recover;
7. Repeat for each commit you want to recover...

After you have recovered all the desired commits, you can create a new branch to keep the recovered commits:
<pre>
git checkout -b recovered-commits
</pre>
and push the recovered-commits branch to your remote repository

I hope this info helps you. Please, let me know and good luck

**Reply by ghost** [reply]
Sadly I have a mac.

**Reply by neutrinomuon** [reply]
Humm, the terminal commands for Git work the same way on Mac as on Linux, so you can still try using the `git fsck --lost-found` command to recover dangling commits on Mac as well. You can just follow the same steps as mentioned earlier for Linux. 

However, it's important to note that data recovery is not always possible, especially if the commits were deleted and overwritten. Try as soon as you can and try to not save anything on your Mac to not overwrite anything.

P.S.: I have also a MAC apple M1

**Reply by ghost** [reply]
Thank you, unfortunately I deleted the commits and that is why all my files are gone. Big lesson learned today.

**Reply by neutrinomuon** [reply]
I understand :(! So, you can try this free open-source project: https://www.cgsecurity.org/wiki/TestDisk . It was first designed for recovering partitions but I know it recovers files. I've used once some time ago.  Let me know if it works

**Reply by ghost** [reply]
Thank you, I checked it out and found some files on there, unfortunately none that are important or worth saving.

**Comment by David-Garcia9919** [comment]
Here's a similar scare a while back where I lost a bunch of files and thought they were gone for good. Here's a similar scare a while back where I lost a bunch of files and thought they were gone for good. I ended up using EdbMails [Windows Data Recovery Software](https://www.edbmails.com/pages/data-recovery-software.html) and it managed to pull back most of what I'd lost. Worth a shot if you haven't tried a recovery tool yet, just stop saving new files to that drive first since that's what kills your odds of recovery.
